contracts/license/src/state.rs

```diff
@@ -59,13 +59,12 @@ impl LicenseContractState {
 impl LicenseContractState {
     /// Inserts a license into the collection of licenses.
     /// Method intended to be called by the License Provider.
-    pub fn issue_license(
-        &mut self,
-        license: Vec<u8>,
-        pos: u64,
-        hash: BlsScalar,
-    ) {
+    pub fn issue_license(&mut self, license: Vec<u8>, hash: BlsScalar) {
         let item = PoseidonItem { hash, data: () };
+        let mut pos = self.tree.len();
+        while self.tree.contains(pos) {
```

Comment:
Why there is the need of cycle? Don't you append always?

Reply:
Cycle is needed in case the position is taken (occupied) by another party running in parallel after the length is read (very unlikely but still a possibility IMO)